server/src/lib/id.ts

```diff
@@ -0,0 +1,9 @@
+import { Request } from "express";
+import crypto from "crypto";
+
+export function getId(req: Request) {
+  const userAgent = req.headers["user-agent"];
+  const ip = req.headers["x-forwarded-for"] || req.connection.remoteAddress;
+  const id = crypto.createHash("md5").update(`${userAgent}${ip}`).digest("hex");
```

Comment:
Another fun FYI, `md5` hashing is now explicitly not recommended for device fingerprinting:
>A drawback of cryptographic hash algorithms such as [MD5](https://en.wikipedia.org/wiki/MD5) and [SHA](https://en.wikipedia.org/wiki/Secure_Hash_Algorithm_(disambiguation)) is that they take considerably longer to execute than Rabin's fingerprint algorithm. They also lack proven guarantees on the collision probability. Some of these algorithms, notably [MD5](https://en.wikipedia.org/wiki/MD5), are no longer recommended for secure fingerprinting. They are still useful for error checking, where purposeful data tampering isn't a primary concern.

Again, it _really_ doesn't matter for this application, but thought you might be interested. For a legit application, the standard would be to use a library version of or implement your own version of "Rabin's Algorithm". Sorry if this not interesting haha, I just spent way too much time on device fingerprinting a couple years back.

Reply:
I think we can pin this as an issue or something and address it in the future.

Reply:
Created #28 just to keep track of this.